src/basic/hash-funcs.c

```diff
@@ -6,7 +6,7 @@
 #include "path-util.h"
 
 void string_hash_func(const char *p, struct siphash *state) {
-        siphash24_compress(p, strlen(p) + 1, state);
+        siphash24_compress_string(p, state);
```

Comment:
hmm, so this is a slight change in behaviour. previously we'd hash the trailing NUL byte too. now we do no longer. I think it would be better to hash it, so that hashing a NUL terminated string has a different result than hashing just a bunch of bytes...

I wonder if we ever use generated hashes in a way that we write them to disk. Because we'd suddenly start hashing differently...

Reply:
OK. This change is dropped.